packages/calcite-components/src/components/table-row/table-row.tsx

```diff
@@ -385,7 +390,11 @@ export class TableRow implements InteractiveComponent, LocalizedComponent {
             aria-rowindex={this.positionAll + 1}
             aria-selected={this.selected}
             class={{ [CSS.lastVisibleRow]: this.lastVisibleRow }}
-            onKeyDown={(event) => this.keyDownHandler(event)}
+            onKeyDown={(event) => {
+              if (this.interactionMode === "interactive") {
```

Comment:
Could we push this into the event handler to avoid creating a new anonymous function on each render?

Reply:
Updated and will merge in post-Chromatic run. Thanks for review!